source/posix/thread.c

```diff
@@ -39,13 +39,14 @@ static void *thread_fn(void *arg) {
     return NULL;
 }
 
-struct aws_thread_options *aws_default_thread_options(void) {
+const struct aws_thread_options *aws_default_thread_options(void) {
```

Comment:
I'm going to point out here that this API is fundamentally incompatible with having ABI forwards compatibility with extending structures with new members - existing code doesn't know how to copy the structure, and the technique of sticking the structure size in the first member won't even work with this.

This is a place where we need to commit to a model of how we handle ABI compat concerns before committing too much to this API, I think.

Reply:
It can be a fair tradeoff IFF we get the thread options right before going GA, since that's not something that changes very much.